### PR TITLE
feat(sql): Add function to allow casts from string to boolean

### PR DESCRIPTION
**Pull request by marregui**
This PR follows https://github.com/questdb/questdb/pull/2000 where a test was ignored because these statements threw exceptions:

```
select cast('' AS BOOLEAN) IS NULL --> false;
select cast('' AS BOOLEAN) IS NOT NULL --> true;
```

Pull 2000 contains tests for these statements (where `<non-nullable-type>` in [boolean, short, byte]):

```
select cast(0 AS <non-nullable-type>) IS NULL --> false;
select cast(0L AS <non-nullable-type>) IS NULL --> false;
select cast('' AS <non-nullable-type>) IS NULL --> false;
select cast(NULL AS <non-nullable-type>) IS NULL --> false;
select cast(false AS <non-nullable-type>) IS NULL --> false;

select cast(0 AS <non-nullable-type>) IS NOT NULL --> true;
select cast(0L AS <non-nullable-type>) IS NOT NULL --> true;
select cast('' AS <non-nullable-type>) IS NOT NULL --> true;
select cast(NULL AS <non-nullable-type>) IS NOT NULL --> true;
select cast(false AS <non-nullable-type>) IS NOT NULL --> true;
```

While I was at it, I decided that you could also cast strings:

```
   'true' -> true::boolean
   'anything other than true, including false' -> false::boolean
```

e.g.

```
cast('tRuE' as BOOLEAN) --> true;
cast('a-banana' as BOOLEAN) --> false;
cast('false' as BOOLEAN) --> false;
```

so that a statement like this is allowed and delivers predictable results:

```
select cast(rnd_str('28', 'TRuE', '', null, 'false', 'true') as boolean) boolean from long_sequence(10);
boolean[BIT]
false
false
false
false
true
false
false
true
false
false
```

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 16 / 16 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/cast/CastStrToBooleanFunctionFactory.java|16|16|100.00%|

**Comment by bluestreak01** [comment]
Could you summarize this PR please, what it fixed and include usable links to other PRs?

File: core/src/test/java/io/questdb/cutlass/pgwire/PGJobContextTest.java [hunk 1]
```diff
@@ -2675,6 +2675,72 @@ public void testInsertFomNodeJsWith2Parameters_WithTableCreation() throws Except
...
+            ) {
+                conn.prepareStatement(
+                        "create table booleans (value boolean, ts timestamp) timestamp(ts)"
+                ).execute();
+
+                Rnd rand = new Rnd();
+                String [] values = {"TrUE", null, "", "false", "true", "banana", "22"};
+
+                try (PreparedStatement insert = conn.prepareStatement("insert into booleans values (cast(? as boolean), ?)")) {
+                    long micros = TimestampFormatUtils.parseTimestamp("2022-04-19T18:50:00.998666Z");
```
Comment (bziobrowski) [inline core/src/test/java/io/questdb/cutlass/pgwire/PGJobContextTest.java:2693]: What's the purpose of randomizing value cast as boolean ?

File: core/src/main/java/io/questdb/griffin/engine/functions/cast/CastStrToBooleanFunctionFactory.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    @Override
+    public Function newInstance(int position, ObjList<Function> args, IntList argPositions, CairoConfiguration configuration, SqlExecutionContext sqlExecutionContext) throws SqlException {
+        Function arg0 = args.getQuick(0);
+        if (arg0.isConstant()) {
+            return resolveBoolean(arg0.getStr(null)) ? BooleanConstant.TRUE : BooleanConstant.FALSE;
+        }
+        return new Func(arg0);
+    }
+
+    private static boolean resolveBoolean(CharSequence str) {
```
Comment (bziobrowski) [inline core/src/main/java/io/questdb/griffin/engine/functions/cast/CastStrToBooleanFunctionFactory.java:55]: Isn't this conditional too complex ? 
Wouldn't something like : 

`return str != null && SqlKeywords.isTrueKeyword(str);`

be a bit more readable ?

File: core/src/test/java/io/questdb/cutlass/pgwire/PGJobContextTest.java (hunk 1 above)
Comment (marregui) [reply core/src/test/java/io/questdb/cutlass/pgwire/PGJobContextTest.java:2693]: to provide an assortment of string candidates to become a boolean value without having to roll out the code into several inserts.

File: core/src/main/java/io/questdb/griffin/engine/functions/cast/CastStrToBooleanFunctionFactory.java (hunk 2 above)
Comment (marregui) [reply core/src/main/java/io/questdb/griffin/engine/functions/cast/CastStrToBooleanFunctionFactory.java:55]: indeed :)

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 12 / 12 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/cast/CastStrToBooleanFunctionFactory.java|12|12|100.00%|

File: core/src/test/java/io/questdb/cutlass/pgwire/PGJobContextTest.java (hunk 1 above)
Comment (bziobrowski) [reply core/src/test/java/io/questdb/cutlass/pgwire/PGJobContextTest.java:2693]: Let me rephrase the question . What does iteration + random give versus simple iteration over values array ? 
Are you trying to trigger some condition in pg wire "state machine" ?

Comment (marregui) [reply core/src/test/java/io/questdb/cutlass/pgwire/PGJobContextTest.java:2693]: ?? nope, it is just my style, what is the smell of clouds?

**Review by bziobrowski — APPROVED** [review]
